Drv/ByteStreamDriverModel/ByteStreamRecvPortAi.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?oxygen RNGSchema="file:../xml/ISF_Type_Schema.rnc" type="compact"?>
+
+<interface name="ByteStreamRecv" namespace="Drv">
+    <include_header>Fw/Buffer/Buffer.hpp</include_header>
+    <args>
+        <arg name="recvBuffer" type="Fw::Buffer" pass_by="reference">
+        </arg>
+        <arg name="recvStatus" type="ENUM" pass_by="value">
+            <enum name="RecvStatus">
+                <item name="RECV_OK" comment="Receive worked as expected" />
+                <item name="RECV_ERROR" comment="Receive error occurred retrying may succeed"/>
```

Comment:
Remember, the bytestreammodel is intended to be the generic driver model for future drivers that can run comm.   Not every driver will be capable of reopening. This is a generic error that future developers can use.

Reply:
My bigger questions is when should a driver ignore errors (and just not send a recv message to the user) vs send a recv message with an error. Is there a byte stream protocol where sending an error makes sense? Off the top of my head maybe a device is suddenly disconnected we could let the user know with a recv error. I guess giving the api the power to express errors is useful.

Reply:
It depends on the driver.  Network blips shouldn't take down an IP stack as users expect reasonable robustness in the face of network problems.  Reconnecting can and does recover from the problem.  So here, the receive thread will try to reconnect as the default behavior, but can be told not to do so.  

A properly wired UART port is rock-solid, and here we report errors instead, because that usually means a non-recoverable problem and reconnecting (typically) does not help.  If I were to implement a UART driver I would not reconnect an report errors, because I believe them to be rarer and re-connections to be less useful. .....except when my USB port isn't fully seated....but that is another story.

Reply:
I do believe that the IP drivers should report an error if re-connection is disabled.  As that would follow the protocol.